components/director/internal/domain/api/repository_test.go

```diff
@@ -2,343 +2,217 @@ package api_test
 	"github.com/kyma-incubator/compass/components/director/internal/domain/api/automock"
 	"github.com/kyma-incubator/compass/components/director/internal/model"
 	"github.com/kyma-incubator/compass/components/director/internal/repo/testdb"
-	"github.com/kyma-incubator/compass/components/director/pkg/persistence"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 )
 
 func TestPgRepository_GetByID(t *testing.T) {
-	// given
-	apiDefEntity := fixFullEntityAPIDefinition(apiDefID, "placeholder")
-
-	selectQuery := fmt.Sprintf(`^SELECT (.+) FROM "public"."api_definitions" WHERE %s AND id = \$2$`, fixTenantIsolationSubquery())
-
-	t.Run("success", func(t *testing.T) {
-		sqlxDB, sqlMock := testdb.MockDatabase(t)
-		rows := sqlmock.NewRows(fixAPIDefinitionColumns()).
-			AddRow(fixAPIDefinitionRow(apiDefID, "placeholder")...)
-
-		sqlMock.ExpectQuery(selectQuery).
-			WithArgs(tenantID, apiDefID).
-			WillReturnRows(rows)
+	entity := fixFullEntityAPIDefinition(apiDefID, "placeholder")
+	apiDefModel, _, _ := fixFullAPIDefinitionModel("placeholder")
 
-		ctx := persistence.SaveToContext(context.TODO(), sqlxDB)
-		convMock := &automock.APIDefinitionConverter{}
-		convMock.On("FromEntity", apiDefEntity).Return(model.APIDefinition{Tenant: tenantID, BaseEntity: &model.BaseEntity{ID: apiDefID}}, nil).Once()
-		pgRepository := api.NewRepository(convMock)
-		// WHEN
-		modelAPIDef, err := pgRepository.GetByID(ctx, tenantID, apiDefID)
-		//THEN
-		require.NoError(t, err)
-		assert.Equal(t, apiDefID, modelAPIDef.ID)
-		assert.Equal(t, tenantID, modelAPIDef.Tenant)
-		convMock.AssertExpectations(t)
-		sqlMock.AssertExpectations(t)
-	})
+	suite := testdb.RepoGetTestSuite{
+		Name: "Get API",
+		SQLQueryDetails: []testdb.SQLQueryDetails{
+			{
+				Query:    regexp.QuoteMeta(`SELECT id, app_id, package_id, name, description, group_name, ord_id, short_description, system_instance_aware, api_protocol, tags, countries, links, api_resource_links, release_status, sunset_date, changelog_entries, labels, visibility, disabled, part_of_products, line_of_business, industry, version_value, version_deprecated, version_deprecated_since, version_for_removal, ready, created_at, updated_at, deleted_at, error, implementation_standard, custom_implementation_standard, custom_implementation_standard_description, target_urls, extensible, successors, resource_hash FROM "public"."api_definitions" WHERE id = $1 AND (id IN (SELECT id FROM api_definitions_tenants WHERE tenant_id = $2))`),
+				Args:     []driver.Value{apiDefID, tenantID},
+				IsSelect: true,
+				ValidRowsProvider: func() []*sqlmock.Rows {
+					return []*sqlmock.Rows{sqlmock.NewRows(fixAPIDefinitionColumns()).AddRow(fixAPIDefinitionRow(apiDefID, "placeholder")...)}
+				},
+				InvalidRowsProvider: func() []*sqlmock.Rows {
+					return []*sqlmock.Rows{sqlmock.NewRows(fixAPIDefinitionColumns())}
+				},
+			},
+		},
+		ConverterMockProvider: func() testdb.Mock {
```

Comment:
The idea here in the generic suits is a bit different. This functions is used as a mock provider. Then in the actual suite each test uses this function to generate itself a mock from the specified type (since tests are generic we need to do define it outside where the suite is created and we have the mock type). Then each test of the suite calls this function to create a mock and then configures the expected invocations based on the test scenario.

In this case this is not an expected unused state and more of a mock generator that can be used to create empty (new) mocks for the specific type.